### DISCUSSION TITLE
Keyball 61　両手トラックボールは可能？

### DISCUSSION THREAD

**Discussion by mako3313** [Yowkees/keyball#549] [Q&A] [answered]
件名の通りです
資料・ディスカッション等を漁ったのですが、両手用ファームウェアに関しては凍結中なのでしょうか？
ご確認のほどよろしくお願いします

**Comment by koron** [comment, marked as answer]
現在のファームウェアで可能です。
([参考ソースコード](https://github.com/Yowkees/keyball/blob/main/qmk_firmware/keyboards/keyball/lib/keyball/keyball.c))

左右のトラックボール有り基板をTRRSケーブルでつないだ状態でRemapに接続すると
そのような状態として認識されているのが確認できるはずです。
([参考ソースコード](https://github.com/Yowkees/keyball/blob/f5665173e073a343132da147f429458be47283cf/qmk_firmware/keyboards/keyball/lib/keyball/keyball.c#L313-L319))

トラックボールの振舞いとしては
USBを接続したほうのトラックボールがマウスポインタの移動で、反対側がスクロールになります。
このあたりはハードコーディングされてるので変更するにはコード修正が必要です。
([参考ソースコード](https://github.com/Yowkees/keyball/blob/f5665173e073a343132da147f429458be47283cf/qmk_firmware/keyboards/keyball/lib/keyball/keyball.c#L267-L268))

両ボールにするとキーがかなり少なくなりますので、
キーマップについては個々人で再検討、再設定してください。
